streamingccl: rename Create and Plan in client interface

msbutler opened the pull request:
This patch renames Create to CreateForTenant and Plan to PlanPhysicalReplication to better destinguish which api calls relate to PCR vs LDR.

Follow up work will move CreateForTenant and PlanPhysicalReplication into its own client interface wrapper for PCR.

Epic: none

Release note: none

msbutler committed:
streamingccl: rename functions in client and logicalReplicationClient interface

This patch renames Create to CreateForTenant and Plan to
PlanPhysicalReplication in Client and PartitionSpans to PlanLogicalReplication
in LogicalReplicationClieng to better destinguish which api calls relate to PCR
vs LDR.

Follow up work will move CreateForTenant and PlanPhysicalReplication into its
own client interface wrapper for PCR.

Epic: none

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/125572)
<!-- Reviewable:end -->

msbutler committed:
streamproducer: rename functions to delineate LDR from PCR

This patch renames crdb_internal.partition_spans to
crdb_internal.plan_logical_replication, and on replicationStreamManager,
PartitionSpans to PlanLogicalReplication and GetReplicationStreamSpec to
GetPhysicalReplicationStreamSpec. I intentionally did not rename
crdb_internal.replication_stream_specover cross version compatability concerns.

Epic: none

Release note: none

msbutler commented:
unrelated flakes

stevendanna commented on pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go:
```diff
@@ -63,8 +63,8 @@ func NewPartitionedStreamClient(
+func (p *partitionedStreamClient) CreateForTenant(
 	ctx context.Context, tenantName roachpb.TenantName, req streampb.ReplicationProducerRequest,
 ) (streampb.ReplicationProducerSpec, error) {
 	ctx, sp := tracing.ChildSpan(ctx, "streamclient.Client.Create")
```
Update span name.

stevendanna commented on pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go:
```diff
@@ -275,7 +275,7 @@ func (p *partitionedStreamClient) Complete(
+func (p *partitionedStreamClient) PlanLogicalReplication(
 	ctx context.Context, spans []roachpb.Span,
 ) (Topology, error) {
 	ctx, sp := tracing.ChildSpan(ctx, "streamclient.Client.PartitionSpans")
```
Update span name.

msbutler replied:
Done.

stevendanna commented on pkg/ccl/streamingccl/streamclient/client.go:
```diff
@@ -118,7 +118,7 @@ type Client interface {
 	Client
 
-	PartitionSpans([]roachpb.Span) (Topology, error)
+	PlanLogicalReplication([]roachpb.Span) (Topology, error)
```
Heh, it doesn't matter because nothing is using this interface.  But this signature is wrong.

msbutler replied:
Done.

msbutler commented:
TFTR!

bors r=stevendanna

msbutler commented:
bors r+
